logger/src/main/java/com/palantir/logsafe/logger/SafeLoggerFactory.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * (c) Copyright 2021 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.logsafe.logger;
+
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeLoggable;
+import com.palantir.logsafe.logger.spi.SafeLoggerFactoryBridge;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.ServiceLoader;
+
+/** Factory used to access {@link SafeLogger} instance. */
+public final class SafeLoggerFactory {
+
+    private static final SafeLoggerFactoryBridge BRIDGE =
+            ServiceLoader.load(SafeLoggerFactoryBridge.class, SafeLoggerFactory.class.getClassLoader()).stream()
+                    .map(ServiceLoader.Provider::get)
+                    .max(Comparator.comparing(SafeLoggerFactoryBridge::priority))
+                    // Should never happen unless dependencies are explicitly excluded
+                    .orElseThrow(NoSafeLoggerImplementationsException::new);
+
+    /** Returns a {@link SafeLogger} for the {@code clazz} origin. */
+    public static SafeLogger get(@Safe Class<?> clazz) {
+        return new SafeLogger(BRIDGE.get(clazz));
+    }
+
+    /** Returns a {@link SafeLogger} for the {@code name} origin. */
+    public static SafeLogger get(@Safe String name) {
```

Comment:
We could enforce compile-time-constant strings here, or remove the string option entirely. I think `@Safe` is sufficient,and haven't encountered unsafe leaks through the logger name myself, so I don't want to make it difficult to implement things like https://github.com/palantir/tritium/blob/b3995307803a97cd2b5e1c5abc1d463070cac181/tritium-annotations/src/main/java/com/palantir/tritium/annotations/internal/InstrumentationBuilder.java#L78-L80
```suggestion
    public static SafeLogger get(@Safe @CompileTimeConstant String name) {
```

Reply:
Agreed that compile time constant here is likely desirable, but not required, and would be a break in a few places. The risk here is that we propagate an Unsafe origin field (which is assumed safe) but we already hold that risk and it has never occurred (and highly unlikely to).